### PR TITLE
Add macOS support

### LINKED ISSUE #5: Support macOS
Many projects may want to run CI on both Linux & macOS, so `install-swift` should support installation on macOS as well.

### PR DESCRIPTION
**Pull request by PoshAlpaca**
## Changes
- Now also supports macOS
- CI now also tests on macOS
- Split script into step-like functions invoked from `install_swift` main function

Closes #5.

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -9,19 +9,37 @@ inputs:
 runs:
   using: "composite"
   steps:
-  - name: Determine Ubuntu Version
+  - name: Determine Cache Key
```
Comment (slashmo) [inline action.yaml]:
```suggestion
  - name: Determine cache options
```

Nitpick, but maybe we also want to indicate here that it's not only about the cache key but also the path.

File: action.yaml [hunk 2]
```diff
@@ -9,19 +9,37 @@ inputs:
...
+
+      if [[ $SWIFT_VERSION == swift-DEVELOPMENT-SNAPSHOT* ]]; then
+          RELEASE_NAME=$SWIFT_VERSION
+      else
+          RELEASE_NAME=swift-$SWIFT_VERSION-RELEASE
+      fi
+
+      if [ $RUNNER_OS == Linux ]; then
+        PLATFORM=ubuntu-$(lsb_release -rs)
+        TOOLCHAIN_PATH=/opt/swift-toolchains/$RELEASE_NAME
```
Comment (slashmo) [inline action.yaml]:
```suggestion
        TOOLCHAIN_CACHE_PATH=/opt/swift-toolchains/$RELEASE_NAME
```

File: action.yaml [hunk 3]
```diff
@@ -9,19 +9,37 @@ inputs:
...
+      elif [ $RUNNER_OS == macOS ]; then
+        PLATFORM=macos-$(sw_vers -productVersion)
+        TOOLCHAIN_PATH=${HOME}/Library/Developer/Toolchains/$RELEASE_NAME.xctoolchain
```
Comment (slashmo) [inline action.yaml]:
```suggestion
        TOOLCHAIN_CACHE_PATH=${HOME}/Library/Developer/Toolchains/$RELEASE_NAME.xctoolchain
```

File: action.yaml [hunk 4]
```diff
@@ -9,19 +9,37 @@ inputs:
...
+      fi
+
+      echo "TOOLCHAIN_PATH=$TOOLCHAIN_PATH" >> $GITHUB_ENV
```
Comment (slashmo) [inline action.yaml]:
```suggestion
      echo "TOOLCHAIN_CACHE_PATH=$TOOLCHAIN_CACHE_PATH" >> $GITHUB_ENV
```

File: action.yaml [hunk 5]
```diff
@@ -9,19 +9,37 @@ inputs:
...
     shell: bash
 
   - name: Cache Swift Toolchain
     id: swift-toolchain-cache
     uses: actions/cache@v2
     with:
-      path: /usr/share/swift-toolchain
-      key: ubuntu-${{ env.UBUNTU_VERSION }}-${{ inputs.version }}
+      path: ${{ env.TOOLCHAIN_PATH }}
+      key: ${{ env.CACHE_KEY }}
```
Comment (slashmo) [inline action.yaml]:
```suggestion
      path: ${{ env.TOOLCHAIN_CACHE_PATH }}
      key: ${{ env.TOOLCHAIN_CACHE_KEY }}
```

File: install-swift.sh [hunk 6]
```diff
@@ -1,54 +1,189 @@
...
-    echo "Toolchain already exists, skipping download ✅"
-else
-    echo "Reading toolchain version from .swift-version 📄"
-    echo "Detected Swift toolchain version '$(cat .swift-version)' 📄"
+# Expects environment variable `SWIFT_VERSION`
+# Example values: 5.5 or swift-DEVELOPMENT-SNAPSHOT-2022-01-09-a
 
+# System library dependencies: wget
+
+install_swift () {
```
Comment (slashmo) [inline install-swift.sh:8]: Nice, this is way cleaner 👍

**Review by slashmo — APPROVED** [review]
LGTM, only a naming nitpick in the action file.

File: action.yaml (hunk 1 above)
Comment (PoshAlpaca) [reply action.yaml]:
```suggestion
  - name: Determine Cache Options
```

File: action.yaml [hunk 7]
```diff
@@ -9,19 +9,37 @@ inputs:
...
+      if [ $RUNNER_OS == Linux ]; then
+        PLATFORM=ubuntu-$(lsb_release -rs)
+        TOOLCHAIN_CACHE_PATH=/opt/swift-toolchains/$RELEASE_NAME
+      elif [ $RUNNER_OS == macOS ]; then
+        PLATFORM=macos-$(sw_vers -productVersion)
+        TOOLCHAIN_CACHE_PATH=${HOME}/Library/Developer/Toolchains/$RELEASE_NAME.xctoolchain
+      fi
+
+      echo "TOOLCHAIN_CACHE_PATH=$TOOLCHAIN_CACHE_PATH" >> $GITHUB_ENV
+      echo "CACHE_KEY=$PLATFORM-$SWIFT_VERSION" >> $GITHUB_ENV
```
Comment (slashmo) [inline action.yaml]:
```suggestion
      echo "TOOLCHAIN_CACHE_KEY=$PLATFORM-$SWIFT_VERSION" >> $GITHUB_ENV
```

**Pull request merged**